### ISSUE TITLE
[Assignment 1] How do I run my Rust code after changing it? (very simple question)

### ISSUE THREAD

**Issue by 1starJ** [kaist-cp/cs220#254]
### Googling Result

https://stackoverflow.com/questions/72715314/run-specific-rs-file

### ChatGPT Result

https://chat.openai.com/share/6a14796c-31bc-4fc1-b7af-89c4e95b6459

### Your question here

```markdown
First of all, excuse me for asking a question that might seem so trivial. I don't have any git experience, and I need some help.
My question is about very basic thing: How do I run a Rust code in an assignement folder of cs220 directory? Should I build a new cargo project for every assignment, or is there more simple way to run the codes in a specific assignment folder?

I'm struggling because in python, I was able to just run a .py file right away. But I know that in Rust, I have to make a new directory by command `cargo new something` and build a project by `cargo build` then I can run the code. 

However, I don't know how to run the .rs file(the code) that is already in a folder. What is a command for running an already existing .rs file? If my question is not right, please explain to me about the process you are going through when doing the assignments.

Things that I have tried: I'm using the Visual Studio Code program (local editor) for programming. I cloned the cs220 directory (using git clone command) to my computer, and got the assignments. Then, I did a little change to the small_exercises.rs by replacing `todo!()` with `return lhs+rhs;`. I went to git bash program and commanded `$ cargo run`, but got an error.
```

**Comment by 1starJ** [comment]
Oh, by the way, some should change the format of "your question here". It seems like a code.

**Comment by minseongg** [comment]
For assignment 1, you can test your code by using `cargo test --lib assignment01`.

**Comment by 1starJ** [comment]
@minseongg Thank you for your consideration. I could successfully test my code.
I have another question. Is there any way to actually see the result of my code directly? Testing the code by `./scripts/grade.sh1` only let you know whether you got the right answer, but not the actual result.

**Comment by minseongg** [comment]
@1starJ Could you please try `cargo test --lib assignment01 -- --nocapture` ?

**Comment by jirheee** [comment]
## TL;DR
- use the development server
- clone the repo to the development server
- process of solving the assignments
  - read the docs and tests
  - write code where we placed to `todo!()`s
  - test them by running the grade script `scripts/grade.sh 01`

## How to test your programs

You should be able to test your program by running the grade script as it is written in the `README.md` of this repository.

```sh
// to run grading script of assignment 1
./scripts/grade.sh 1
```

By running the script, it will automatically run all the relavant tests(by `cargo test`) and show your score.

If you want to know how this works, I recommend reading the `scripts/grading.sh` script, and learning about the bash scripting. Learning about the shell scripting  will help you a lot through your life as a CS major, since lot of grading scripts and automations are made with it😎

One thing to note is that the grading script is likely to work as intended in the development server(which uses Linux as its operating system). So we insist you to use the development server provided to minimize the troubleshooting irrelevant to the course.

If you struggle to use the development server, please let us know! We are always here to help you.

## Where are the tests?

Also, all the tests that you should pass are in the same directory of the assignment.

```
assignment01
|- mod.rs
|- some_assignment.rs
|- some_assignment_grade.rs
..
```

To learn more about how all this works, about how modules work and how test works, I recommend reading:
- [Rust modules by example](https://doc.rust-lang.org/rust-by-example/mod.html)
- [`cargo test` by example](https://doc.rust-lang.org/rust-by-example/cargo/test.html)

## Sidenote

I know learning all this different tools like git, linux, ssh, shell scripting and etc at the first time, especially as a first year of computer science major(I assume you are a second grader since this course is cs2**) can be very frustrating 🥲 

It was same for me when I first took CS course. Nobody told me what tools to use, I started with putty, vim and notepad while taking systems programming course, and it was a very bad experience. But after struggling through some amount of time, I learned the tools, and they turned out to be very useful(at least as a programmer), and I am still using them as a grads student! I bet it will be same for you, if you choose your career as something related to programming.

And for last(sorry for being so long,,,😅), you don't have to be sorry for any question. We are here to help you learn and go through the course, not to judge you! So feel free to ask any questions. We will do our best to help you.

**Comment by 1starJ** [comment]
Thank you so much! It helped me a lot.
